packages/react/src/createInertiaApp.ts

```diff
@@ -82,6 +84,10 @@ export default async function createInertiaApp<SharedProps extends PageProps = P
 
   let head = []
 
+  if (!isServer) {
+    setupModal({ showOnNonInertiaResponse: showModalOnNonInertiaResponse })
```

Comment:
also, if we wanted to follow the same "API" used for the progress configs, we could instead have a `modal = {}` props and then call `setupModal(modal)`

then

Reply:
A usage example would then look like:
 
 ```jsx
 createInertiaApp({
  modal: {
    showOnNonInertiaResponse: false
  },
  progress: { 
    delay: 300,
    color: '#29d'
  },
  resolve: (name) => import(`./Pages/${name}`),
  setup({ el, App, props }) {
    render(<App {...props} />, el)
  }
})
```